turbopack/crates/turbopack-ecmascript/src/references/pattern_mapping.rs

```
@@ -83,7 +94,7 @@ pub(crate) enum PatternMapping {
     /// ```js
     /// require(`./images/${name}.png`)
     /// ```
-    Map(FxIndexMap<String, SinglePatternMapping>),
+    Map(#[bincode(with = "turbo_bincode::indexmap")] FxIndexMap<String, SinglePatternMapping>),
```

This annotation will have to be specified for every occurrence of indexmap?

---

Unfortunately, yes.

If you don't want something like this, you need to implement the `Encode`/`Decode` traits on `IndexMap`. Because of the orphan rules, the only other options are that either `bincode` needs to take a dependency on `indexmap`, or `indexmap` needs to take a dependency on `bincode`.

That's not too unreasonable: often times libraries will provide this behind a feature. `indexmap` provides a `serde` feature that does this.

I might try to do this in bincode behind a feature if I have success in getting my existing set of patches (which include adding the `with` feature) upstreamed to bincode. I've already submitted one patch via email.

---

But you do get an error message when forgetting this? Or does it silently use the wrong serialization impl

It's fine, not much we can do apart from trying to upstream.

---

You get a compile-time error. The `Encode`/`Decode` derive macros fail to compile.